hpcbench/driver.py

```diff
@@ -401,12 +401,28 @@ def _execute_sbatch(self):
         commands = self.campaign.process.get('commands', {})
         sbatch = find_executable(commands.get('sbatch', 'sbatch'))
         sbatch_command = [sbatch, '--parsable', self.sbatch_filename]
-        sbatch_out = subprocess.check_output(sbatch_command,
-                                             universal_newlines=True)
-        jobid = sbatch_out.split(';')[0].strip()
-        LOGGER.info("Submitted SBATCH job %s for tag %s",
-                    jobid, self.tag)
-        return int(jobid)
+        try:
+            sbatch_out = subprocess.check_output(sbatch_command,
+                                                 universal_newlines=True)
+        except subprocess.CalledProcessError as cpe:
+            LOGGER.error("SBATCH return non-zero exit status %d for tag %s",
+                         cpe.returncode, self.tag)
+            sbatch_out = cpe.output
+        jobidre = re.compile('^([\d]+)(?:;\S*)?$')
+        jobid = None
+        for line in sbatch_out.split('\n'):
+            res = jobidre.match(line)
+            if res is not None:
+                jobid = res.group(1)
+                LOGGER.info("Submitted SBATCH job %s for tag %s",
+                            jobid, self.tag)
+            else:
```

Comment:
IMHO, we should only log `sbatch_out` when `jobid` cannot be extracted instead of log a warning for every line that do not match the regular expression.

Furthermore, would it be possible to add a test demonstrating the fix based on tds output?

Reply:
I thought it would be useful to put other output as warnings, because when all went well actually the --parsable flag of sbatch should not return anything but the jobid, so when you actually get some output it is noteworthy..

yeah.. i knew you'd ask about a special test.. it would entail changing the test_slurm setup/teardown to also produce an sbatch mock that "breaks". Not a problem, but it seemed a bit overkill for this case.. 
also, keep in mind that our next window for testing hpcbench in a more serious setting is approaching fast.